### PR TITLE
Handle plain object body in "Send Email" operation

### LINKED ISSUE #16842: Flow: Mail not sent when the body contains only {{$trigger.payload}}
### Describe the Bug

When the body is only `{{$trigger.payload}}` the mail is not sent.

### To Reproduce

Send a mail from a flow only `Details: {{$trigger.payload}}` in the body. The mail is sent.
Modify the body of the mail to only `{{$trigger.payload}}` in the body. 

_Not very important, but may hide some other potential bug_

### Errors Shown

The mail is not sent when reduced to `{{$trigger.payload}}`

### What version of Directus are you using?

9.21.0

### What version of Node.js are you using?

19.0.1

### What database are you using?

Postgresql 15

### What browser are you using?

Firefox & Edge

### How are you deploying Directus?

VPS

### PR DESCRIPTION
**Pull request by paescuj**
## Description

If body is of type object / undefined in the "Send Email" operation (happens when body consists solely of a placeholder) convert it to JSON string so it can be processed correctly.

Not entirely sure if this applies to other operations as well and should thus be solved in the `FlowManager` itself.
Also not sure whether it would be better to convert `null` & `undefined` to an empty string.
But since it is rather an edge case, I think this fix is fine for now.

Fixes #16842 
Fixes ENG-306

## Type of Change

- [x] Bugfix
- [ ] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

### CODE REVIEW

File: api/src/operations/mail/index.ts [hunk 1]
```diff
@@ -14,9 +14,11 @@ export default defineOperationApi<Options>({
 
 	handler: async ({ body, to, type, subject }, { accountability, database, getSchema }) => {
 		const mailService = new MailService({ schema: await getSchema({ database }), accountability, knex: database });
-
+		// If 'body' is of type object/undefined (happens when body consists solely of a placeholder)
+		// convert it to JSON string
+		const safeBody = typeof body === 'object' || typeof body === 'undefined' ? JSON.stringify(body) : body;
```
Comment (br41nslug) [inline api/src/operations/mail/index.ts:19]: While `object`/`null` covers most bases here the mailservice will still fail when being passed a `Number` directly (possibly for a `Function` too)a.
```suggestion
		const safeBody = typeof body !== 'string' ? JSON.stringify(body) : body;
```

**Review by br41nslug — CHANGES_REQUESTED** [review]

**Comment by azrikahar** [comment]
#16773 might be addressing the undefined-to-JSON-string issue as well 🧐

File: api/src/operations/mail/index.ts (hunk 1 above)
Comment (rijkvanzanten) [reply api/src/operations/mail/index.ts:19]: Thanks @br41nslug! I like that suggestion 👍🏻

**Review by rijkvanzanten — APPROVED** [review]

**Comment by rijkvanzanten** [comment]
@azrikahar It fixes undefined on the overarching flows level. This PR ensures that the outgoing email body is always a string 👍🏻

**Review by br41nslug — APPROVED** [review]

**Pull request merged**